chain/actors/builtin/miner/actor.go.template

```diff
@@ -319,3 +332,37 @@ func AllCodes() []cid.Cid {
     {{- end}}
     }
 }
+
+func PledgePenaltyForContinuedFault(
+		nwVer network.Version,
+		rewardEstimate builtin.FilterEstimate,
+		networkQaPowerEstimate builtin.FilterEstimate,
+		qaSectorPower abi.StoragePower,
+) (abi.TokenAmount, error) {
+	v, err := actorstypes.VersionForNetwork(nwVer)
+	if err != nil {
+		return big.Zero(), err
+	}
+	switch v {
+	{{- range .versions}}
+	case actorstypes.Version{{.}}:
+		{{- if (ge . 16)}}
+			return minertypes{{.}}.PledgePenaltyForContinuedFault(
+				smoothing{{.}}.FilterEstimate{
```

Comment:
later: we should consider moving `FilterEstimate` to a shared package.

Reply:
yep, on my list